src/App.tsx

```diff
@@ -1,9 +1,28 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  let visibleMovies;
+
+  if (moviesFromServer.find(movie => (
+    movie.title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+  ))) {
+    visibleMovies = moviesFromServer.filter(movie => (
+      movie.title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+    ));
+  }
+
+  if (moviesFromServer.find(movie => (
+    movie.description.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+  ))) {
+    visibleMovies = moviesFromServer.filter(movie => (
+      movie.description.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+    ));
+  }
```

Comment:
too much going on here:
- you don't need this checks before filtering, you just go through array 2 times
- you can combine both conditions in 1 filter function
- you repeat yourself with a `query.toLocaleLowerCase()` a lot